sw/device/lib/testing/test_rom/test_rom_start.S

```diff
@@ -187,6 +188,17 @@ _start:
   li   t0, 0x55aa
   sw   t0, EDN_CTRL_REG_OFFSET(a0)
 
+  // Scramble and initialize main memory (main SRAM).
+  // Memory accesses will stall until initialization is complete.
+  // Skip SRAM initialization for DV sim device type, as the testbench handles
+  // this to optimize test run times.
+  lw    t0, kDeviceType
+  beqz  t0, .L_sram_init_skip
+  li    a0, TOP_EARLGREY_SRAM_CTRL_MAIN_REGS_BASE_ADDR
+  li    a1, (1 << SRAM_CTRL_CTRL_RENEW_SCR_KEY_BIT) | (1 << SRAM_CTRL_CTRL_INIT_BIT)
+  sw    a1, SRAM_CTRL_CTRL_REG_OFFSET(a0)
```

Comment:
Not that it should matter, but it looks like the convention in the test_rom is to use a0 and t0.

Does the test rom need to RENEW_SCR_KEY?  I omitted this bit from my test.

Reply:
naw. If you don't tell it to renew key, it simply uses the default key, which also works in this case.

Reply:
I don't think it should need to renew the scr key, I was just keeping consistent with mask ROM. Skipping this will save ~800 cycles according to the docs, so I will remove... and fix the a0/t0 convention to match :)

Reply:
fixed.